### PR TITLE
Remove support for Python 2

### PR DESCRIPTION
**Pull request by henri-hulski**
This PR removes support for Python 2 as we decided in https://github.com/morepath/reg/issues/55#issuecomment-511098204.

### CODE REVIEW

**Comment by href** [comment]
👍

File: doc/conf.py [hunk 1]
```diff
@@ -125,7 +125,7 @@
 # Add any paths that contain custom static files (such as style sheets) here,
 # relative to this directory. They are copied after the builtin static files,
 # so a file named "default.css" will overwrite the builtin "default.css".
-html_static_path = ['_static']
+html_static_path = []
```
Comment (faassen) [inline doc/conf.py]: This seems unrelated to the Python 3 update at first sight. I am not sure _static was doing anything, though?

File: reg/arginfo.py [hunk 2]
```diff
@@ -93,18 +94,15 @@ class Dummy(object):
...
 
 def get_class_init(class_):
-    try:
-        func = class_.__init__
-    except AttributeError:  # pragma: no cover
-        # Python 2 classic class without __init__.
-        return fake_empty_init
+    func = class_.__init__
+
     # If this is a new-style class and there is no __init__
```
Comment (faassen) [inline reg/arginfo.py]: I was hoping that the differences between PyPy 3 and Python 3.7 were less, but the code to handle this is still in place. Does this mean this is still needed?

File: setup.cfg [hunk 3]
```diff
@@ -1,6 +1,6 @@
 [flake8]
 show-source = True
-ignore = E731
+ignore = E731, W503
```
Comment (faassen) [inline setup.cfg]: A lot of these changes seem to be completely unrelated to Python 3, but change the style of the binary operators to be after the line instead of before. That's why you shut off this warning. Can we please focus this PR on Python 3 support and not unrelated coding style changes?

In the future I want to move the code style to black so we don't have this discussion anymore.

Comment (henri-hulski) [reply setup.cfg]: The problem is that either W503 or W504 is triggering and W504 seems to be the correct one.

https://lintlyci.github.io/Flake8Rules/rules/W503.html
https://lintlyci.github.io/Flake8Rules/rules/W504.html

File: doc/conf.py (hunk 1 above)
Comment (henri-hulski) [reply doc/conf.py]: Not really related to Python 3 but I also tried to fix the tests and I got an error of a non-existing '_static' directory in 'src'. As we don't use this directory we should also remove it from the configuration.

File: setup.cfg (hunk 3 above)
Comment (faassen) [reply setup.cfg]: All right, I withdraw my objection and we'll worry about it again when we have black.

File: reg/arginfo.py (hunk 2 above)
Comment (faassen) [reply reg/arginfo.py]: I would like us to look into this before we merge it. But if the tests do work we can instead choose to merge it anyway and look into it later (create a github issue for Reg if you decide to do that). So I'm approved in principle.

**Comment by faassen** [comment]
I'm okay with a merge as-is. We'll resolve the styling issues once we introduce Black. I *would* like us to look into whether the PyPy specific code is still needed - my hope was that PyPy 3 is more similar to Python 3 and we don't need custom logic anymore.

File: reg/arginfo.py (hunk 2 above)
Comment (henri-hulski) [reply reg/arginfo.py]: @faassen I have removed the PyPy specific code. Also removed some PyPy code in Importscan (importscan/scan.py).
